src/sources/socket/mod.rs

```diff
@@ -1157,11 +1436,38 @@ mod test {
         .await;
     }
 
+    #[cfg(unix)]
+    #[tokio::test]
+    async fn unix_stream_message_with_log_namespace() {
+        assert_source_compliance(&SOCKET_HIGH_CARDINALITY_PUSH_SOURCE_TAGS, async {
+            let (_, rx) = unix_message("test", true, true).await;
+            let events = collect_n(rx, 1).await;
+            let event_meta = events[0].as_log().metadata().value();
+
+            assert_eq!(1, events.len());
+            assert_eq!(
+                event_meta.get(path!("vector", "source_type")).unwrap(),
+                &vrl::value!(SocketConfig::NAME)
+            );
+            assert_eq!(
+                events[0].as_log()[log_schema().message_key()],
```

Comment:
In the vector namespace, this isn't dependent on the `.message_key()` but is determined by the codec.

Reply:
afaik `log_schema().message_key()` is always used in the code to refer to the default values for these fields. Do you think we should move away from that pattern here/

Reply:
`log_schema()` should only be pointing to something in the legacy namespace, I think this works now because the default decoder for socket happens to decode into `.message` field still.

A couple examples:
https://github.com/vectordotdev/vector/blob/master/lib/codecs/src/decoding/format/bytes.rs#L79-L84
Vector namespace: the message is literally the root of the event, rather than being an object

https://github.com/vectordotdev/vector/blob/master/lib/codecs/src/decoding/format/syslog.rs#L155-L157
Vector namespace: the event is an object, and the field used for the message "happens" to have the same name as the old default `.message_key()`

Reply:
Gotcha. Thanks for the feedback!